### PR TITLE
[ADD] fieldservice_stock_location_builder

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR overrides the create_sub_locations() method in FSMLocationBuilderWizard so all sublocations have the same inventory location as the parent location.

Depends on #326

#129

### CODE REVIEW

**Review by patrickrwilson — CHANGES_REQUESTED** [review]
Works well, will approve once all green.

**Comment by layadelcarmen** [comment]
I have made a functional review of this module. The wizard creates the new locations with the parent inventory location. Wizard lines needs some validation, admit empty level name and odoo throw an error .`    name + spacer + str(lev_id),
TypeError: unsupported operand type(s) for +: 'bool' and 'str'`

File: fieldservice_stock_location_builder/wizard/fieldservice_location_builder_wizard.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                        level_ids[num].name + spacer + str(lev_id),
+                        'owner_id': location.owner_id.id,
+                        'customer_id': location.customer_id.id,
+                        'fsm_parent_id': parent.id,
+                        'inventory_location_id': location.
+                        inventory_location_id.id}
+                new_location = self.env['fsm.location'].create(vals)
+                if num < levels:
+                    build_location(new_location, num + 1)
+        build_location(location, 0)
```
Comment (max3903) [inline fieldservice_stock_location_builder/wizard/fieldservice_location_builder_wizard.py]: @osi-scampbell I would add a method `prepare_fsm_location_values()` in `fieldservice_location_builder` so you can extend it here just by adding the inventory location to the dictionary.

**Comment by max3903** [comment]
Ping @osi-scampbell

File: fieldservice_stock_location_builder/__manifest__.py [hunk 2]
```diff
@@ -10,7 +10,8 @@
     'website': 'https://github.com/OCA/field-service',
     'depends': [
         'fieldservice_stock',
-        'fieldservice_location_builder'
+        'fieldservice_location_builder',
+        'fieldservice_account'
```
Comment (max3903) [inline fieldservice_stock_location_builder/__manifest__.py]: @osi-scampbell Why do you need `fieldservice_account`?

Comment (osi-scampbell) [reply fieldservice_stock_location_builder/__manifest__.py]: That is where we moved customer_id to

Comment (max3903) [reply fieldservice_stock_location_builder/__manifest__.py]: @osi-scampbell The `customer_id` field on `fsm_location` is added in `fieldservice_account_analytic`.

1. Let's remove the dependency here
2. In `fieldservice_account_analytic`, add a default value on the field with a method:
```
if self.parent_id:
    return self.parent_id.customer_id.id
return self.owner_id.id
```

**Comment by osi-scampbell** [comment]
#474

**Pull request closed without merging**